contracts/MultiSend.sol

```diff
@@ -1,74 +1,117 @@
+// Copyright (C) 2018 Alon Bukai This program is free software: you 
+// can redistribute it and/or modify it under the terms of the GNU General 
+// Public License as published by the Free Software Foundation, version. 
+// This program is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details. You should have received a copy of the GNU General Public
+// License along with this program. If not, see http://www.gnu.org/licenses/
+
 pragma solidity ^0.4.18;
 
 import "giveth-common-contracts/contracts/Escapable.sol";
 
-// TightlyPacked is cheaper if you need to store input data and if amount is less than 12 bytes.
-// Normal is cheaper if you don't need to store input data or if amounts are greater than 12 bytes.
-// Supports deterministic deployment. As explained here: https://github.com/ethereum/EIPs/issues/777#issuecomment-356103528
+/// @notice MultiSend is a contract for sending multiple ETH/ERC20 Tokens to multiple addresses
+/// In addition this contract can call multiple contracts with multiple amounts
+/// There are also TightlyPacked functions which in some situations allow for gas savings
 contract MultiSend is Escapable {
   
+  // TightlyPacked is cheaper if you need to store input data and if amount is less than 12 bytes.
+  // Normal is cheaper if you don't need to store input data or if amounts are greater than 12 bytes.
+  // Supports deterministic deployment. As explained here: https://github.com/ethereum/EIPs/issues/777#issuecomment-356103528
+
+
+  /// @notice Hardcoded escapeHatchCaller
   address CALLER = 0x839395e20bbB182fa440d08F850E6c7A8f6F0780;
+  /// @notice Hardcoded escapeHatchDestination
   address DESTINATION = 0x8ff920020c8ad673661c8117f2855c384758c572;
 
+  /// @notice Constructor using Escapable and Hardcoded values
   function MultiSend() Escapable(CALLER, DESTINATION) public {}
   
-  function multiTransferTightlyPacked(bytes32[] _addressAndAmount) payable public returns(bool) {
-    for (uint i = 0; i < _addressAndAmount.length; i++) {
-        _safeTransfer(address(_addressAndAmount[i] >> 96), uint(uint96(_addressAndAmount[i])));
+  /// @notice Send to multiple addresses using a byte32 array which
+  /// includes the address and the amount.
+  /// Payable
+  /// @param _addressesAndAmounts Bitwise packed array of addresses and amounts
```

Comment:
Try and include some information on where the address is and where the amounts are. The address is stored in the 20 most significant bytes of the byte32 object and the amount is stored in the 12 least significant bytes of the byte32 object.